sycl/test/warnings/warnings.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -Wall -Wpessimizing-move -Wunused-variable -Wmismatched-tags -Wunneeded-internal-declaration -Werror -Wno-unknown-cuda-version -fsycl %s -o %t.out
+// RUN: %clangxx -fsycl -I %source_sycl_include -Wall -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wpessimizing-move -Wunused-variable -Wmismatched-tags -Wunneeded-internal-declaration -Werror -Wno-unknown-cuda-version  %s -o %t.out
```

Comment:
`-I %source_sycl_include` - why do you need this?
AFAIK, we copy SYCL headers to build directory, which is included by default. What is wrong with these headers?
If something is wrong with them, it sounds like a bug to me, which should be fixed.

Reply:
`-fsycl` automatically adds include path to SYCL headers, but they are assumed to be a "system" once, which means that no warnings are emitted from them.

Therefore, we need to pass include path to them explicitly to make them "user" once.

If I recall correctly, experiments showed that passing path to a build directory doesn't do the trick and we need to pass path into a directory place somewhere outside of default location

Reply:
First, you should learn the proper way to enable diagnostics for system headers: https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-in-system-headers

Second, 

> If I recall correctly, experiments showed that passing path to a build directory doesn't do the trick and we need to pass path into a directory place somewhere outside of default location

if this is the case, it's a bug, which must be fixed.

Reply:
> First, you should learn the proper way to enable diagnostics for system headers: https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-in-system-headers

This is actually good to know, thanks!
@fadeeval, let's update the test to use correct way of enabling diagnostics

> > If I recall correctly, experiments showed that passing path to a build directory doesn't do the trick and we need to pass path into a directory place somewhere outside of default location
>
> if this is the case, it's a bug, which must be fixed.

@fadeeval, could you please prepare a simple reproducer, perform an experiment again and if the behavior is still as I described, file a bug?

Reply:
I made issue https://github.com/intel/llvm/issues/1732